kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -402,41 +489,56 @@ define([
-                        templateSelection: selectionTemplate,
-                        ajax: {
-                            delay: 250,
-                            transport: function (params, success, failure) {
-                                return fetchData(params.data.term)
-                                    .then((data) => {
-                                        success({ results: data });
-                                    })
-                                    .catch((err) => {
-                                        console.error(err);
-                                        failure(err);
-                                    });
-                            },
+                const dropdown = $(ui.getElement('input-container.input'));
+                dropdown.select2({
+                    allowClear: true,
+                    templateResult: formatObjectDisplay,
+                    templateSelection: selectionTemplate,
+                    ajax: {
+                        delay: 250,
+                        processResults: (data) => {
+                            // update the currently selected one if applicable
+                            const currentData = dropdown.select2('data')[0];
+                            if (currentData) {
+                                const updatedData = data.results.filter((item) => {
+                                    return item.id === currentData.id;
+                                });
+                                if (updatedData.length) {
+                                    for (const key of Object.keys(updatedData[0])) {
+                                        currentData[key] = updatedData[0][key];
+                                    }
+                                    dropdown.trigger('change');
+                                }
+                            }
+                            return data;
                         },
-                        placeholder: {
-                            id: 'select an option',
+                        transport: function (params, success, failure) {
+                            return fetchData(params.data.term)
+                                .then((data) => {
+                                    success({ results: data });
+                                })
+                                .catch((err) => {
+                                    console.error(err);
+                                    failure(err);
+                                });
                         },
-                    })
+                    },
+                    placeholder: {
+                        id: 'select an option',
+                    },
+                });
+                dropdown
                     .on('change', () => {
-                        doChange();
+                        const data = dropdown.select2('data');
+                        return doChange(data ? data[0] : {});
```

Comment:
Send the data object along when doing a change, so we save the display value as well.